File: NotesApplication.Client/src/types/note.ts
export interface Note {
  id: number
  title: string
  content: string | null
  createdAt: string
  updatedAt: string | null
  userId?: string
}

export interface CreateNoteDto {
  userId: number
  title: string
  content?: string | null
}

export interface UpdateNoteDto {
  userId: number
  title: string
  content?: string | null
}

export type SortField = 'title' | 'createdAt' | 'updatedAt'
export type SortDirection = 'asc' | 'desc'

export interface SortOption {
  field: SortField
  direction: SortDirection
  label: string
}

export interface NotesQueryState {
  search: string
  sortField: SortField
  sortDirection: SortDirection
}

File: NotesApplication.API/Models/Note.cs
using System;

namespace NotesApplication.API.Models
{
    public class Note
    {
        public int NoteId { get; set; }

        public int UserId { get; set; }

        public string Title { get; set; } = string.Empty;

        public string? Content { get; set; }

        public DateTime CreatedAt { get; set; }

        public DateTime? UpdatedAt { get; set; }
    }
}